src/torchpme/potentials/potential_dipole.py

```diff
@@ -0,0 +1,134 @@
+from typing import Optional
+
+import torch
+
+from .potential import Potential
+
+
+class PotentialDipole(torch.nn.Module):
+    """TODO: Add docstring"""
+
+    def __init__(
+        self,
+        smearing: Optional[float] = None,
+        exclusion_radius: Optional[float] = None,
+        dtype: Optional[torch.dtype] = None,
+        device: Optional[torch.device] = None,
+    ):
+        super().__init__()
+        if dtype is None:
+            dtype = torch.get_default_dtype()
+        if device is None:
+            device = torch.device("cpu")
+        if smearing is not None:
+            self.register_buffer(
+                "smearing", torch.tensor(smearing, device=device, dtype=dtype)
+            )
+        else:
+            self.smearing = None
+        if exclusion_radius is not None:
+            self.register_buffer(
+                "exclusion_radius",
+                torch.tensor(exclusion_radius, device=device, dtype=dtype),
+            )
+        else:
+            self.exclusion_radius = None
+
+    @torch.jit.export
+    def f_cutoff(self, vector: torch.Tensor) -> torch.Tensor:
+        r"""TODO: Add docstring"""
+        r_mag = torch.norm(vector, dim=1)
+        if self.exclusion_radius is None:
+            raise ValueError(
+                "Cannot compute cutoff function when `exclusion_radius` is not set"
+            )
+
+        return torch.where(
+            r_mag < self.exclusion_radius,
+            (1 + torch.cos(r_mag * (torch.pi / self.exclusion_radius))) * 0.5,
+            0.0,
+        )
+
+    def from_dist(self, vector: torch.Tensor) -> torch.Tensor:
+        """TODO: Add docstring"""
+        r_mag = torch.norm(vector, dim=1, keepdim=True)
+        scalar_potential = 1.0 / (r_mag**3)
+        r_outer = torch.einsum(
+            "bi,bj->bij", vector, vector
```

Comment:
Might be worth for speed later to replace the einsum by outer. The latter is usually much faster.